src/lm/extlm_chainer.py

```diff
@@ -104,13 +104,22 @@ def __call__(self, state, x):
             else:
                 wlm_logprob = wlm_logprobs[:, self.word_unk] + self.log_oov_penalty
             log_y[:, self.space] = wlm_logprob
-            log_y[:, self.eos] = wlm_logprob + wlm_logprobs[:, self.word_eos]
+            log_y[:, self.eos] = wlm_logprob
```

Comment:
Why did you change like that? I just don't fully understand this part.

Reply:
Actually, `<eos>` probability was incorrect here although ASR accuracy was the same before and after this change.
The LM state needed to be updated with the last word to get `<eos>` prob, but it is computationally very expensive. So I made `final()` method in the word RNNLM classes to compute `<eos>` prob, and changed the beam search code to apply `<eos>` prob only for `ended_hyps`.